applications/FluidDynamicsApplication/custom_elements/embedded_fluid_element.cpp

```diff
@@ -0,0 +1,494 @@
+{
+    return Element::Pointer(new EmbeddedFluidElement(NewId, pGeom, pProperties));
+}
+
+template <class TBaseElement>
+void EmbeddedFluidElement<TBaseElement>::CalculateLocalSystem(
+    MatrixType& rLeftHandSideMatrix, VectorType& rRightHandSideVector,
+    ProcessInfo& rCurrentProcessInfo) {
+
+    // Resize and intialize output
+    if (rLeftHandSideMatrix.size1() != LocalSize)
+        rLeftHandSideMatrix.resize(LocalSize, LocalSize, false);
+
+    if (rRightHandSideVector.size() != LocalSize)
+        rRightHandSideVector.resize(LocalSize, false);
+
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize, LocalSize);
+    noalias(rRightHandSideVector) = ZeroVector(LocalSize);
+
+    EmbeddedElementData data;
+    data.Initialize(*this, rCurrentProcessInfo);
+    this->InitializeGeometryData(data);
+
+    // Iterate over integration points on the volume
+    const unsigned int number_of_positive_gauss_points =
+        data.PositiveSideWeights.size();
+    for (unsigned int g = 0; g < number_of_positive_gauss_points; g++) {
+        data.UpdateGeometryValues(data.PositiveSideWeights[g],
+            row(data.PositiveSideN, g), data.PositiveSideDNDX[g]);
+
+        this->AddTimeIntegratedSystem(
+            data, rLeftHandSideMatrix, rRightHandSideVector);
+    }
+
+    if ( data.IsCut() ) {
+        // Iterate over integration points on the boundary
+        const unsigned int number_of_interface_gauss_points =
+            data.PositiveInterfaceWeights.size();
+        for (unsigned int g = 0; g < number_of_interface_gauss_points; g++) {
+            data.UpdateGeometryValues(data.PositiveInterfaceWeights[g],
+                row(data.PositiveInterfaceN, g), data.PositiveInterfaceDNDX[g]);
+            this->AddBoundaryIntegral(data, data.PositiveInterfaceUnitNormals[g],
+                rLeftHandSideMatrix, rRightHandSideVector);
+        }
+
+        // First, compute and assemble the penalty level set BC imposition contribution
+        // Secondly, compute and assemble the modified Nitche method level set BC imposition contribution (Codina and Baiges, 2009)
+        // Note that the Nistche contribution has to be computed the last since it drops the outer nodes rows previous constributions
+        AddBoundaryConditionPenaltyContribution(rLeftHandSideMatrix, rRightHandSideVector, data);
+        DropOuterNodesVelocityContribution(rLeftHandSideMatrix, rRightHandSideVector, data);
+        AddBoundaryConditionModifiedNitscheContribution(rLeftHandSideMatrix, rRightHandSideVector, data);
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Inquiry
+
+template <class TBaseElement>
+int EmbeddedFluidElement<TBaseElement>::Check(
+    const ProcessInfo& rCurrentProcessInfo) {
+
+    EmbeddedElementData::Check(*this,rCurrentProcessInfo);
```

Comment:
The point is that all checks I have seen so far return 0. If something doesn't work, they just throw an error. I can check the return type, but it is more future-proofing than a real issue at this point. Maybe we should reconsider if we really need error codes, since we (as far as I know) always use exceptions?